stl/inc/deque

```diff
@@ -1219,6 +1219,31 @@ public:
         return begin() + static_cast<difference_type>(_Off);
     }
 
+private:
+    enum class _Pop_direction : bool {
+        _Front,
+        _Back,
+    };
+
+    template <_Pop_direction _Direction>
+    struct _NODISCARD _Restore_old_size_guard {
+        deque* _Container;
+        const size_type _Oldsize;
+

```

Comment:
Okay, I'm willing to let this one go since this is a very localized guard class. Deleting a constructor would make the class non-aggregrate, so we'd also need to add a constructor which would be provide an annoying debugger experience.